src/ast/passes/codegen_llvm.cpp

```diff
@@ -745,10 +746,14 @@ void CodegenLLVM::visit(Call &call)
     str_id_++;
     expr_ = buf;
   } else if (call.func == "buf") {
-    Value *max_length = b_.getInt64(
-        bpftrace_.config_.get(ConfigKeyInt::max_strlen));
-    size_t fixed_buffer_length = bpftrace_.config_.get(
-        ConfigKeyInt::max_strlen);
+    uint64_t max_strlen = bpftrace_.config_.get(ConfigKeyInt::max_strlen);
+    // Eventually we codegen a u32 store, so ensure we are within limits
+    assert(max_strlen <= std::numeric_limits<uint32_t>::max());
+
+    // Subtract out metadata headroom
+    uint64_t max_buffer_size = max_strlen - sizeof(AsyncEvent::Buf);
```

Comment:
why not just name this variable `fixed_buffer_length` instead of creating a new one below?